cmd/run.go

```diff
@@ -142,6 +148,31 @@ Run sidecar only:
 					os.Exit(1)
 				}
 
+				if appPort <= 0 {
+					// If app does not listen to port, we can check for Dapr's sidecar health before starting the app.
+					// Otherwise, it creates a deadlock.
+					sidecarUp := true
+					print.InfoStatusEvent(os.Stdout, "Checking if Dapr sidecar is listening on HTTP port %v", output.DaprHTTPPort)
+					err = utils.IsDaprListeningOnPort(output.DaprHTTPPort, time.Duration(runtimeWaitTimeoutInSeconds)*time.Second)
+					if err != nil {
+						sidecarUp = false
+						print.WarningStatusEvent(os.Stdout, "Dapr sidecar is not listening on HTTP port: %s", err.Error())
+					}
+
+					print.InfoStatusEvent(os.Stdout, "Checking if Dapr sidecar is listening on GRPC port %v", output.DaprGRPCPort)
```

Comment:
the same code block is running twice, move it to a loop and break from loop once side car is up. So that InfoStatusEvent is not printed twice and IsDaprListeningOnPort is not called again even if the first one succeeds

Reply:
This is a different port. This is for GRPC.

Reply:
oh ok, thanks for clarifying.